scripts/src/javatests/org/oppia/android/scripts/maven/GenerateMavenDependenciesListTest.kt

```diff
@@ -440,33 +514,119 @@ class GenerateMavenDependenciesListTest {
+    )
+    val license1ForDependency2 = dependency2.licenseList[0]
+    verifyLicenseHasExtractedCopyVerifiedLink(
+      license = license1ForDependency2,
+      originalLink = "",
+      licenseName = "Simplified BSD License",
+      verifiedLink = "https://local-copy/bsd-license"
+    )
+  }
+
+  private fun verifyLicenseHasScrapableVerifiedLink(
+    license: License,
+    originalLink: String,
+    licenseName: String,
+    verifiedLink: String,
+  ) {
+    val scrapableLink = ScrapableLink.newBuilder().setUrl(verifiedLink).build()
+    assertThat(license.licenseName).isEqualTo(licenseName)
+    assertThat(license.scrapableLink).isEqualTo(scrapableLink)
+    assertThat(license.originalLink).isEqualTo(originalLink)
+    assertThat(license.isOriginalLinkInvalid).isEqualTo(false)
+  }
+
+  private fun verifyLicenseHasExtractedCopyVerifiedLink(
+    license: License,
+    originalLink: String,
+    licenseName: String,
+    verifiedLink: String,
+  ) {
+    val extractedCopyLink = ExtractedCopyLink.newBuilder().setUrl(verifiedLink).build()
+    assertThat(license.licenseName).isEqualTo(licenseName)
+    assertThat(license.extractedCopyLink).isEqualTo(extractedCopyLink)
+    assertThat(license.originalLink).isEqualTo(originalLink)
+    assertThat(license.isOriginalLinkInvalid).isEqualTo(false)
+  }
+
+  private fun verifyLicenseHasDirectLinkOnlyVerifiedLink(
+    license: License,
+    originalLink: String,
+    licenseName: String,
+    verifiedLink: String,
+  ) {
+    val directLinkOnly = DirectLinkOnly.newBuilder().setUrl(verifiedLink).build()
+    assertThat(license.licenseName).isEqualTo(licenseName)
+    assertThat(license.directLinkOnly).isEqualTo(directLinkOnly)
+    assertThat(license.originalLink).isEqualTo(originalLink)
+    assertThat(license.isOriginalLinkInvalid).isEqualTo(false)
+  }
+
+  private fun verifyLicenseHasVerifiedLinkNotSet(
+    license: License,
+    originalLink: String,
+    licenseName: String
+  ) {
+    assertThat(license.licenseName).isEqualTo(licenseName)
+    assertThat(license.verifiedLinkCase).isEqualTo(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET)
+    assertThat(license.originalLink).isEqualTo(originalLink)
+    assertThat(license.isOriginalLinkInvalid).isEqualTo(false)
+  }
+
+  private fun verifyLicenseHasOriginalLinkInvalid(
+    license: License,
+    originalLink: String,
+    licenseName: String
+  ) {
+    assertThat(license.licenseName).isEqualTo(licenseName)
+    assertThat(license.verifiedLinkCase).isEqualTo(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET)
+    assertThat(license.originalLink).isEqualTo(originalLink)
+    assertThat(license.isOriginalLinkInvalid).isEqualTo(true)
+  }
+
+  private fun assertDependency(
+    dependency: MavenDependency,
+    artifactName: String,
+    artifactVersion: String,
+  ) {
+    assertThat(dependency.artifactName).isEqualTo(artifactName)
+    assertThat(dependency.artifactVersion).isEqualTo(artifactVersion)
+  }
+
+  private fun getProto(
```

Comment:
Suggest: ``parseTextProto``?

also for consistency, it may be nice to do a quick find & replace for 'textproto' to be two words (e.g. s/Textproto/TextProto & s/textproto/textProto & s/textproto/text proto) depending on the circumstances.

Reply:
Done.

Reply:
Is this completed? I still see ``getProto``.

Reply:
done.

Reply:
Done.